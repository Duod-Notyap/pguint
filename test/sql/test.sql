SELECT '55'::uint2;
SELECT '70000'::uint2;
SELECT ''::uint2;
SELECT '-55'::uint2;


CREATE TABLE test2 (a uint2);

INSERT INTO test2 VALUES (1), (2), (3), (4), (5);

SELECT a FROM test2;


SELECT '55'::uint4;
SELECT ''::uint4;
SELECT '-55'::uint4;


CREATE TABLE test4 (a uint4);

INSERT INTO test4 VALUES (1), (2), (3), (4), (5);

SELECT a FROM test4;

SELECT avg(a) FROM test4;

SELECT sum(a) FROM test4;
