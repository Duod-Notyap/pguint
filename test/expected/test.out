SELECT '55'::uint2;
 uint2 
-------
 55
(1 row)

SELECT '70000'::uint2;
ERROR:  value "70000" is out of range for type uint2
LINE 1: SELECT '70000'::uint2;
               ^
SELECT ''::uint2;
ERROR:  invalid input syntax for unsigned integer: ""
LINE 1: SELECT ''::uint2;
               ^
SELECT '-55'::uint2;
ERROR:  invalid input syntax for unsigned integer: "-55"
LINE 1: SELECT '-55'::uint2;
               ^
CREATE TABLE test2 (a uint2);
INSERT INTO test2 VALUES (1), (2), (3), (4), (5);
SELECT a FROM test2;
 a 
---
 1
 2
 3
 4
 5
(5 rows)

SELECT '55'::uint4;
 uint4 
-------
 55
(1 row)

SELECT ''::uint4;
ERROR:  invalid input syntax for unsigned integer: ""
LINE 1: SELECT ''::uint4;
               ^
SELECT '-55'::uint4;
ERROR:  invalid input syntax for unsigned integer: "-55"
LINE 1: SELECT '-55'::uint4;
               ^
CREATE TABLE test4 (a uint4);
INSERT INTO test4 VALUES (1), (2), (3), (4), (5);
SELECT a FROM test4;
 a 
---
 1
 2
 3
 4
 5
(5 rows)

SELECT avg(a) FROM test4;
        avg         
--------------------
 3.0000000000000000
(1 row)

SELECT sum(a) FROM test4;
 sum 
-----
  15
(1 row)

